### DISCUSSION TITLE
Not working!

### DISCUSSION THREAD

**Discussion by blackmamba08** [moralmunky/Home-Assistant-Mail-And-Packages#891] [Q&A]
Hello, 

Inbox (gmail) is connected, sensors seem to be working, dashboard card works.

However, the hyperlink on the card take me to amazon.com but I am in Canada. I have amazon, walmart, canada post , ups and purolator setup. 

Today I got an email from amazon that my package was shipped. Eagerly waited for the card to update but it did not. I can confirm that I did re-configuration and Mail and Packages can see all of my gmail inboxes which means the integration is fine. It checks it every 30 minutes

I assume this is because I am in Canada with amazon.ca - is there any way I can change that? If so, a small tutorial would be great.

**Comment by firstof9** [comment]
> Today I got an email from amazon that my package was shipped.

All packages only increment the "in transit" count when they're out for delivery. For Amazon that'll be the date listed in the first email they sent you.

**Comment by blackmamba08** [comment]
Since my post, I have ordered 2 new things from Amazon and the card has not updated. When I click the Amazon:0 link it takes me to amazon.com 

How can I troubleshoot this issue? 

It is definitely checking my inbox every 30 mins or so. I can see it in the card history

**Reply by firstof9** [reply]
The Amazon sensor doesn't increment until the day of delivery.

**Comment by appsfinder2** [comment]
> Hello,
> 
> Inbox (gmail) is connected, sensors seem to be working, dashboard card works.
> 
> However, the hyperlink on the card take me to amazon.com but I am in Canada. I have amazon, walmart, canada post , ups and purolator setup.
> 
> Today I got an email from amazon that my package was shipped. Eagerly waited for the card to update but it did not. I can confirm that I did re-configuration and Mail and Packages can see all of my gmail inboxes which means the integration is fine. It checks it every 30 minutes
> 
> I assume this is because I am in Canada with amazon.ca - is there any way I can change that? If so, a small tutorial would be great.

did you add this sensor?
mail_deliveries_message: check in Home Assistant dev

- platform: template
  sensors:
    mail_deliveries_message:
      friendly_name: "Deliveries Summary"
      value_template: > 
        {# Deliveries Sentence #}
          {% macro deliveries_sentence() -%}
                {%- if states("sensor.mail_usps_mail")|int(0) == 0 -%}
                  No
                {%- else -%}
                  {{states("sensor.mail_usps_mail")|int(0)}}
                {%- endif -%}
              {{' '}} 
                {%- if states("sensor.mail_usps_mail")|int(0) <= 1 -%}
                  pieces of mail
                {%- else -%}
                  pieces of mail
                {%- endif -%}
              {{' '}}will be delivered.{{' '}} 
                {%- if states("sensor.mail_usps_delivering")|int(0) == 0 -%}
                  No
                {%- else -%}
                  {{states("sensor.mail_usps_delivering")|int(0)}}
                {%- endif -%}
              {{' '}} 
                {%- if states("sensor.mail_usps_delivering")|int(0) == 1 -%}
                  USPS package is
                {%- else -%}
                  USPS packages are
                {%- endif -%}
              {{' '}}in transit.{{' '}}
                {%- if states("sensor.mail_fedex_delivering")|int(0) == 0 -%}
                  No
                {%- else -%}
                  {{states("sensor.mail_fedex_delivering")|int(0)}}
                {%- endif -%}
              {{' '}} 
                {%- if states("sensor.mail_fedex_delivering")|int(0) == 1 -%}
                  FedEx package is
                {%- else -%}
                  Fedex packages are
                {%- endif -%}
              {{' '}}in transit.{{' '}}
              {%- if states("sensor.mail_ups_delivering")|int(0) == 0 -%}
                  No
                {%- else -%}
                  {{states("sensor.mail_ups_delivering")|int(0)}}
                {%- endif -%}
              {{' '}} 
                {%- if states("sensor.mail_ups_delivering")|int(0) == 1 -%}
                  UPS package is
                {%- else -%}
                  UPS packages are
                {%- endif -%}
              {{' '}}in transit.{{' '}}
              {%- if states("sensor.mail_amazon_packages")|int(0) == 0 -%}
                  No
                {%- else -%}
                  {{states("sensor.mail_amazon_packages")|int(0)}}
                {%- endif -%}
              {{' '}} 
                {%- if states("sensor.mail_amazon_packages")|int(0) == 1 -%}
                  Amazon package is
                {%- else -%}
                  Amazon packages are
                {%- endif -%}
              {{' '}}in transit.{{' '}}
          {%- endmacro %}
        {{deliveries_sentence()}}

**Reply by Manbearpig92** [reply]
yeah I still cant get it to work either I am at the same spot as buddy mentioned before and i added this sensor but i cant get any of them to update when the folder gets mail added to it: 

```
 mail_deliveries_message:
        friendly_name: "Deliveries Summary"
        value_template: >
          {%- macro deliveries_sentence() -%}
            {%- set parts = [] -%}
            
            {# USPS Mail (delivered to your mailbox) #}
            {%- set usps_mail = states("sensor.mail_usps_mail")|int(0) -%}
            {%- if usps_mail > 0 -%}
              {%- set parts = parts + [ usps_mail|string + ( " piece of mail will be delivered" if usps_mail == 1 else " pieces of mail will be delivered" ) ] -%}
            {%- endif -%}
            
            {# USPS packages in transit #}
            {%- set usps_delivering = states("sensor.mail_usps_delivering")|int(0) -%}
            {%- if usps_delivering > 0 -%}
              {%- set parts = parts + [ usps_delivering|string + ( " USPS package is in transit" if usps_delivering == 1 else " USPS packages are in transit" ) ] -%}
            {%- endif -%}
            
            {# FedEx packages in transit #}
            {%- set fedex_delivering = states("sensor.mail_fedex_delivering")|int(0) -%}
            {%- if fedex_delivering > 0 -%}
              {%- set parts = parts + [ fedex_delivering|string + ( " FedEx package is in transit" if fedex_delivering == 1 else " FedEx packages are in transit" ) ] -%}
            {%- endif -%}
            
            {# UPS packages in transit #}
            {%- set ups_delivering = states("sensor.mail_ups_delivering")|int(0) -%}
            {%- if ups_delivering > 0 -%}
              {%- set parts = parts + [ ups_delivering|string + ( " UPS package is in transit" if ups_delivering == 1 else " UPS packages are in transit" ) ] -%}
            {%- endif -%}
            
            {# Amazon packages in transit #}
            {%- set amazon_packages = states("sensor.mail_amazon_packages")|int(0) -%}
            {%- if amazon_packages > 0 -%}
              {%- set parts = parts + [ amazon_packages|string + ( " Amazon package is in transit" if amazon_packages == 1 else " Amazon packages are in transit" ) ] -%}
            {%- endif -%}
            
            {# Canada Post packages in transit – using our custom sensor #}
            {%- set canada_post = states("sensor.canada_post_mail_count")|int(0) -%}
            {%- if canada_post > 0 -%}
              {%- set parts = parts + [ canada_post|string + ( " Canada Post package is in transit" if canada_post == 1 else " Canada Post packages are in transit" ) ] -%}
            {%- endif -%}
            
            {# DHL packages in transit #}
            {%- set dhl_delivering = states("sensor.mail_dhl_delivering")|int(0) -%}
            {%- if dhl_delivering > 0 -%}
              {%- set parts = parts + [ dhl_delivering|string + ( " DHL package is in transit" if dhl_delivering == 1 else " DHL packages are in transit" ) ] -%}
            {%- endif -%}
            
            {# Intelcom packages in transit #}
            {%- set intelcom_delivering = states("sensor.mail_intelcom_delivering")|int(0) -%}
            {%- if intelcom_delivering > 0 -%}
              {%- set parts = parts + [ intelcom_delivering|string + ( " Intelcom package is in transit" if intelcom_delivering == 1 else " Intelcom packages are in transit" ) ] -%}
            {%- endif -%}
            
            {# Purolator packages in transit #}
            {%- set purolator_delivering = states("sensor.mail_purolator_delivering")|int(0) -%}
            {%- if purolator_delivering > 0 -%}
              {%- set parts = parts + [ purolator_delivering|string + ( " Purolator package is in transit" if purolator_delivering == 1 else " Purolator packages are in transit" ) ] -%}
            {%- endif -%}
            
            {# Join all the parts together #}
            {{ parts | join(', ') }}.
          {%- endmacro %}
          {{ deliveries_sentence() }}



      canada_post_mail_count:
        friendly_name: "Canada Post Mail Count"
        value_template: >
          {% set email_text = state_attr('sensor.mail_canada_post_packages', 'body') %}
          {% if email_text %}
            {# This regex looks for a phrase like "You have 7 pieces of mail" #}
            {% set match = email_text | regex_search('You have\\s+(\\d+)\\s+pieces of mail', 1) %}
            {{ match | int if match else 0 }}
          {% else %}
            0
          {% endif %}
```

**Reply by firstof9** [reply]
Check your `home-assistant.log` for errors.

**Reply by Manbearpig92** [reply]
there are not any errors or warnings related to mail but i did find an error in the regex string so i fixed that and well see if it works

**Comment by Manbearpig92** [comment]
https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/issues/124?utm_source=chatgpt.com

okay i think i see the issue, from the post above they provided you an outdated email. canada post has recently changed subject line: "You have mail on the way" and its from: 

donotreply-nepasrepondre@communications.canadapost-postescanada.ca

**Reply by firstof9** [reply]
No that email address is searched for:

https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/blob/61b8ba60990118d677bd3706898f3c5f57ef6f54/custom_components/mail_and_packages/const.py#L280-L292

We're only able to parse the delivered emails.

**Reply by Manbearpig92** [reply]
Hmm, I do have a filter that labels that email address and puts it in "mail" folder/label on Gmail. The subject line is different than what Canada Post is sending from that email address. Would I need to tweak the code to accommodate that on my instance?

**Reply by firstof9** [reply]
What's the subject line for when they've delivered a package?

**Reply by Manbearpig92** [reply]
They actually don't have a notification for delivered that I'm aware of. I haven't gotten it. So far it's just this. But I'm not 100% sure what the arriving soon is in this picture. ![Screenshot_20250212_003406_Gmail.jpg](https://github.com/user-attachments/assets/f21806de-894f-40fa-aeff-dcbf1d848446)

And the subject line is " You have mail on the way" for the email I got from above

**Reply by Manbearpig92** [reply]
![Screenshot_20250212_003156_Gmail.jpg](https://github.com/user-attachments/assets/4ef9b082-73a5-455a-bd96-4aeb1cd35a77)

**Reply by firstof9** [reply]
Canada Post mail isn't supported right now, only packages.

**Reply by Manbearpig92** [reply]
Ah I see, should I make a new post for a feature request or is it already in the works?

**Comment by firstof9** [comment]
Can you check to see if there's a plain text version?

If you're using gmail it's the 3 dots menu -> Show Original

**Comment by Manbearpig92** [comment]
[canada post.json](https://github.com/user-attachments/files/18769489/canada.post.json)


let me know if you need anything else

**Reply by firstof9** [reply]
I'll see what I can wrangle together.

**Reply by firstof9** [reply]
Support added to [0.4.3-b6](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/releases/tag/0.4.3-b6) please give it a test.

**Reply by Manbearpig92** [reply]
okay I will be honest, i have never done this, so i just renamed my old folder mail_and_packages to mail_and_packages_backup and uploaded the whole zip file and restarted. is that all i need to do? i keep getting an error when i try to reconfigure the new canada post mail " invalid boolean value None"

**Reply by firstof9** [reply]
betas should be installed via HACS like normal releases.

**Reply by Manbearpig92** [reply]
Okay so the sensor is working great I just need to wait till I get home if the values are correct because I'm not sure what the time difference is between the out for delivery and arriving soon. I will update you soon

**Comment by Audielite** [comment]
The Amazon is no longer updating either. It used to display how many packages were delivered and a pic of the delivery. Now, nothing. I tried reconfiguring also to no avail.

**Reply by firstof9** [reply]
Looks like they changed the email format, there's PR in to fix it in #1114
